// ngx-prevent-double-submission/index.js
import { Directive, ElementRef, HostListener, NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';

var PreventDoubleSubmitDirective = (function () {
    /**
     * @param {?} el
     */
    function PreventDoubleSubmitDirective(el) {
        this.el = el;
    }
    /**
     * @param {?} $event
     * @return {?}
     */
    PreventDoubleSubmitDirective.prototype.onClick = function ($event) {
        if (!$event['detail'] || $event['detail'] == 1)
            return true;
        else {
            $event.stopPropagation();
            return false;
        }
    };
    return PreventDoubleSubmitDirective;
}());
PreventDoubleSubmitDirective.decorators = [
    { type: Directive, args: [{
                selector: '[prevent-double-submit]',
                exportAs: 'preventDoubleSubmit'
            },] },
];
/**
 * @nocollapse
 */
PreventDoubleSubmitDirective.ctorParameters = function () { return [
    { type: ElementRef, },
]; };
PreventDoubleSubmitDirective.propDecorators = {
    'onClick': [{ type: HostListener, args: ['click', ['$event'],] },],
};

var PreventDoubleSubmitModule = (function () {
    function PreventDoubleSubmitModule() {
    }
    /**
     * @return {?}
     */
    PreventDoubleSubmitModule.forRoot = function () {
        return {
            ngModule: PreventDoubleSubmitModule,
            providers: []
        };
    };
    return PreventDoubleSubmitModule;
}());
PreventDoubleSubmitModule.decorators = [
    { type: NgModule, args: [{
                imports: [
                    CommonModule
                ],
                declarations: [
                    PreventDoubleSubmitDirective
                ],
                exports: [
                    PreventDoubleSubmitDirective
                ]
            },] },
];
/**
 * @nocollapse
 */
PreventDoubleSubmitModule.ctorParameters = function () { return []; };

export { PreventDoubleSubmitModule, PreventDoubleSubmitDirective };
